.travis_scripts/maybe-install-node.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+# Installs correct node and yarn versions if we aren't doing a docker build.
+
+if [ -z "${IMAGE}" ] ; then
+    # install our own yarn to make things work on osx
+    curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version 1.10.1
```

Comment:
Is this as sketchy and insecure as it looks? Can we enforce a hash match on that shell script? Add the script to our own repo?

Or is this realistically no worse than every operation performed by npm or yarn?

Reply:
Semi-sketchy and pre-existing. https://github.com/yarnpkg/website/blob/master/install.sh it checks the GPG key of the tarball it downloads, but obviously if the script itself is compromised we're just executing stuff.